test/VaultOUSDLvUSD.js

```diff
@@ -0,0 +1,172 @@
+        expect(await vaultToken.getVaultOUSDBalance()).to.equal(balance + amount)
+    };
+
+    async function depositOUSDFundsInVaultFromAccount(fromAccount, amount) {
+        // first set it to 0 to mitigate potential abuse https://ethereum.org/en/developers/tutorials/erc20-annotated-code/
+        await OUSDMockToken.connect(fromAccount).approve(vaultToken.address, 0);
+        await OUSDMockToken.connect(fromAccount).approve(vaultToken.address, amount);
+        // console.log("test:Vault:depositFunds:addr1 allowance %s", await)
+        await vaultToken.depositOUSD(fromAccount.address, amount);
+    };
+
+    async function checkVaultOUSDBalance(expected) {
+        let vaultBalance = await vaultToken.getVaultOUSDBalance();
+        expect(vaultBalance).to.equal(expected);
+        expect(await OUSDMockToken.balanceOf(vaultToken.address)).to.equal(expected)
+    };
+
+    async function depositLvUSDIntoVault(amount) {
+        await lvUSDToken.connect(addr1).approve(vaultToken.address, 0);
+        await lvUSDToken.connect(addr1).approve(vaultToken.address, amount);
+        await vaultToken.depositLvUSD(addr1.address, amount);
+    }
+
+    async function checkVaultLvUSDBalance(expected) {
+        expect(await lvUSDToken.balanceOf(vaultToken.address)).to.equal(expected)
+        expect(await vaultToken.getVaultLvUSDBalance()).to.equal(expected);
+
+    }
+
+    describe("Deposit OUSD funds", function () {
+        it("Should be able to deposit OUSD into vault", async function () {
+            let depositedAmount = 1000;
+            console.log("test:Vault:depositFunds:contract address", vaultToken.address);
+            await depositOUSDFundsInVault(depositedAmount)
+            await checkVaultOUSDBalance(depositedAmount)
+        });
+
+        it("Should revert with transfer amount exceeds balance when trying to transfer more then client owns", async function () {
+            // getting current user balance and adding 1 (to exceed balance and fail)
+            let depositedAmount = await OUSDMockToken.balanceOf(addr1.address) + 1;
+
+            // store vault balance before trying to deposit
+            let vaultBalance = await vaultToken.getVaultOUSDBalance()
+
+            // User tries to deposit more OUSD than they have - should revert
+            await expect(depositOUSDFundsInVault(depositedAmount)).to.be.revertedWith("ERC20: transfer amount exceeds balance")
+
+            // we don't expect balance to change, because nothing happened
+            await checkVaultOUSDBalance(vaultBalance)
+        });
+
+        it("Should be able to deposit OUSD in two different transactions", async function () {
+            // mint more OUSD for a different account
+            await OUSDMockToken.mint(addr2.address, defaultUserOUSDBalance)
+
+            let vaultBalance = await vaultToken.getVaultOUSDBalance()
+
+            // perform multiple deposits 
+            await depositOUSDFundsInVaultFromAccount(addr1, 250);
```

Comment:
nitpicking - maybe we should be explicit about decimals. For example 250 could be really $250 (250 *10^18)